python/cudf/cudf/io/parquet.py

```diff
@@ -411,6 +412,26 @@ def read_parquet(
             filepaths_or_buffers.append(tmp_source)
 
     if engine == "cudf":
+        # Temporary error to probe a parquet file
+        # and raise decimal128 support error.
+        if len(filepaths_or_buffers) > 0:
+            try:
+                metadata = pq.read_metadata(filepaths_or_buffers[0])
```

Comment:
Is the schema flat?

Reply:
Pushed the changes to handle nested schema types too.